### ISSUE TITLE
Kodi18 - Wrong inputstream.adaptive version

### ISSUE THREAD

**Issue by Testato** [xbmc/xbmc#19461]
If you install Kodi18 by this command on Ubuntu:
`sudo apt install kodi=2:18.9+git20201024.0821-final-0focal`

it install a wrong version of IA. 
It install the ver 2.6.7 (the new version for Kodi19) instead of ver 2.4.6 (the last working version for Kodi18)

**Comment by phunkyfish** [comment]
@howie-f, maybe you could take a look?

**Comment by howie-f** [comment]
sure thing, looks like a third party repo at a first glance, not our ppa.
@Testato which repository do you pull this from?

**Comment by Testato** [comment]
```
apt-cache policy kodi
kodi:
  Installed: 2:18.9+git20201024.0821-final-0focal
  Candidate: 2:19.0+git20210219.1315-f44fdfbf67-0+final~focal
  Version table:
     2:19.0+git20210219.1315-f44fdfbf67-0+final~focal 500
        500 http://ppa.launchpad.net/team-xbmc/ppa/ubuntu focal/main amd64 Packages
        500 http://ppa.launchpad.net/team-xbmc/ppa/ubuntu focal/main i386 Packages
 *** 2:18.9+git20201024.0821-final-0focal 500
        500 http://ppa.launchpad.net/team-xbmc/kodi-old/ubuntu focal/main amd64 Packages
        500 http://ppa.launchpad.net/team-xbmc/kodi-old/ubuntu focal/main i386 Packages
        500 http://ppa.launchpad.net/team-xbmc/ppa/ubuntu focal/main amd64 Packages
        500 http://ppa.launchpad.net/team-xbmc/ppa/ubuntu focal/main i386 Packages
        100 /var/lib/dpkg/status
     2:18.6+dfsg1-2ubuntu1 500
        500 http://it.archive.ubuntu.com/ubuntu focal/universe amd64 Packages

```

**Comment by wsnipex** [comment]
you forced apt to install an older version of kodi, so you cannot expect that kodi-inputstream-adaptive in it's default(newest) version will work. You have to install an appropriate version for kodi v18.

**Comment by Testato** [comment]
But normally if I install Kodi18 i found in the Official Kodi Repo only Kodi18 compatible plugin.
I have some myself Plugin in the official Kodi repo and I have different version for Kodi18 and Kodi19.
Why this is not true for IA ?

**Comment by wsnipex** [comment]
hint: remove the stable ppa, if you want to use kodi-old

**Comment by howie-f** [comment]
thanks. yes i can confirm

**Comment by Testato** [comment]
But I installed only Kodi18, I not installed IA manually.
Why the Kodi18 installation bring itself automatically the kodi19 version of IA ?

**Comment by wsnipex** [comment]
if you didn't remove the stable ppa yet, run `apt policy kodi-inputstream-adaptive` and see for yourself

**Comment by Testato** [comment]
Also the correct 2.4.6 version is present, so the question remain: Why installing Kodi18 will not install the correct Kodi18 IA ?

```
apt policy kodi-inputstream-adaptive
kodi-inputstream-adaptive:
  Installed: 2.4.6-1~focal
  Candidate: 2.6.7-1~focal
  Version table:
     2.6.7-1~focal 500
        500 http://ppa.launchpad.net/team-xbmc/ppa/ubuntu focal/main amd64 Packages
 *** 2.4.6-1~focal 500
        500 http://ppa.launchpad.net/team-xbmc/kodi-old/ubuntu focal/main amd64 Packages
        100 /var/lib/dpkg/status


```

**Comment by howie-f** [comment]
i think the question is valid. but i‘m not the ppa expert. just can confirm the latest version of IA is installed. it could be by design.. but not really happy from a user pov.

**Comment by wsnipex** [comment]
this really isn't the place to learn how dpkg/apt works, as it has nothing to do with kodi.
tldr: you have added 2 PPAs that contain kodi and IA, v18 in kodi-old and v19 in (stable) ppa.

apt will **by default** (no version given) always install the **highest** available version of any package - from **any of the apt sources**. Furthermore, there is no automatic version downgrade, so once IA 2.6.7 is installed, it won't be downgraded automatically.

As said above: if you want kodi v18, then you really should not have the stable ppa added!

**Comment by Testato** [comment]
I agree that if manually we install IA, without specify the version, it is by design that the last IA is installad.

But if, like I do, install only Kodi18 (and I specify the version of Kodi18) I expect Kodi18 to have dependencies to respect, if one of the needs of Kodi18 is not to exceed the 4.2.6 IA release, can't the Kodi18 team set a limit on the IA version?  New versions of IA will never be compatible with Kodi18.  
Or at least can Kodi18 not install IA, so the user will install the ones that he want.
Automatically install wrong version is worst that not install nothing.

Thanks for the clarifications
